src/s3fs_xml.cpp

```diff
@@ -229,12 +230,13 @@ static unique_ptr_xmlChar get_exp_value_xml(xmlDocPtr doc, xmlXPathContextPtr ct
     return exp_value;
 }
 
-bool get_incomp_mpu_list(xmlDocPtr doc, incomp_mpu_list_t& list)
+std::optional<incomp_mpu_list_t> get_incomp_mpu_list(xmlDocPtr doc)
```

Comment:
I don't think C++17 has major advancements compared to C++11 but `std::optional`  will make error checking more obvious and safer.  These kinds of little enhancements must wait until after we release 1.95 in #2496.